app/models/work.rb

```diff
@@ -609,11 +595,14 @@ def validate_uploads
       end
     end
 
-    # This needs to be called #before_save
```

Comment:
I believe that this still needs to be called in before_save for everything to work correctly.  This was a hint I left myself for later, since I tested and it was required. Maybe both comments?